relay.go

```diff
@@ -654,6 +666,70 @@ func (r *Relayer) handleLocalCallReq(cr *lazyCallReq) bool {
 	return true
 }
 
+func (r *Relayer) fragmentingSend(call RelayCall, f *lazyCallReq, relayToDest relayItem, origID uint32) error {
+	// TODO(echung): should we pool the writers?
+	fragWriter := newFragmentingWriter(
+		r.logger, r.newFragmentSender(relayToDest.destination, f, origID, call),
+		f.checksumType.New(),
+	)
+	defer r.conn.opts.FramePool.Release(f.Frame)
+
+	if len(f.arg2Appends) > 0 && f.isArg2Fragmented {
+		return errFragmentedArg2WithAppend
+	}
+
+	arg2Writer, err := fragWriter.ArgWriter(false /* last */)
+	if err != nil {
+		return fmt.Errorf("get arg2 writer: %v", err)
+	}
+
+	if err := writeArg2WithAppends(arg2Writer, f.arg2(), f.arg2Appends); err != nil {
+		return fmt.Errorf("write arg2: %v", err)
+	}
+	if err := arg2Writer.Close(); err != nil {
+		return fmt.Errorf("close arg2 writer: %v", err)
+	}
+
+	if err := NewArgWriter(fragWriter.ArgWriter(true /* last */)).Write(f.arg3()); err != nil {
+		return errors.New("arg3 write failed")
+	}
+
+	return nil
+}
+
+func writeArg2WithAppends(w io.WriteCloser, arg2 []byte, appends []relay.KeyVal) (err error) {
+	if len(arg2) < 2 {
+		return errNoNHInArg2
+	}
+
+	writer := typed.NewWriter(w)
+
+	// nh:2 is the first two bytes of arg2, which should always be present
+	nh := binary.BigEndian.Uint16(arg2[:2]) + uint16(len(appends))
+	writer.WriteUint16(nh)
```

Comment:
we're using `WriteUint16` here but manually encode to `sizeBuf` and write below? seems like we might be losing the optimization here?

if we end up converting to uint16 manually here, then the typed writer probably isn't adding much value. We may be better off writing to the `[]byte` directly